### PR TITLE
Remove typed client usage

### PR DESCRIPTION
**Pull request by shashidharatd**
This is in continuation of https://github.com/kubernetes-sigs/federation-v2/pull/604. wherever possible, used the generic client in place cluster-registry client and kube clients. Also removed generated unit tests for api's which use generated federation client. they seems redundant.

/cc @marun

### CODE REVIEW

File: pkg/controller/federatedcluster/controller.go [hunk 1]
```diff
@@ -88,13 +80,11 @@ func StartClusterController(config *util.ControllerConfig, stopChan <-chan struc
 
 // newClusterController returns a new cluster controller
 func newClusterController(config *util.ControllerConfig, clusterMonitorPeriod time.Duration) (*ClusterController, error) {
-	client, kubeClient, crClient := config.AllClients("cluster-controller")
+	client := genericclient.NewForConfigOrDieWithUserAgent(config.KubeConfig, "cluster-controller")
```
Comment (marun) [inline pkg/controller/federatedcluster/controller.go]: (No action required) Shades of java-land naming?  ;)

File: pkg/kubefed2/util/util.go [hunk 2]
```diff
@@ -96,8 +96,8 @@ func ClusterClientset(config *rest.Config) (*kubeclient.Clientset, error) {
 
 // ClusterRegistryClientset provides a cluster registry API compliant
 // clientset to communicate with the cluster registry.
-func ClusterRegistryClientset(config *rest.Config) (*crclient.Clientset, error) {
-	return crclient.NewForConfig(config)
+func ClusterRegistryClientset(config *rest.Config) (generic.Client, error) {
```
Comment (marun) [inline pkg/kubefed2/util/util.go]: (No action required) What's the advantage of maintaining this method vs just replacing it with calls to `generic.New`?

Comment (shashidharatd) [reply pkg/kubefed2/util/util.go]: No advantage, just that code change was big. It can be taken up as next step in simplifying the code.

**Comment by shashidharatd** [comment]
> As an aside, are you intending to implement my suggestion of removing the requirement to pass a context to methods in the client interface?

Yes @marun, would consider it in next code optimisation.

**Pull request merged**